packages/comet-admin/src/mui/menu/CollapsibleItem.tsx

```diff
@@ -1,68 +1,89 @@
 import { Collapse, List } from "@material-ui/core";
-import { ListProps } from "@material-ui/core/List";
 import ArrowDownIcon from "@material-ui/icons/KeyboardArrowDown";
 import ArrowUpIcon from "@material-ui/icons/KeyboardArrowUp";
+import { createStyles, WithStyles, withStyles } from "@material-ui/styles";
 import * as React from "react";
+import { matchPath, useLocation } from "react-router";
 
-import { IMenuItemProps, MenuItem } from "./Item";
+import { MenuItem, MenuItemProps } from "./Item";
+import { MenuItemRouterLinkProps } from "./ItemRouterLink";
 
-export interface IMenuLevel {
-    level?: number;
+export type CometAdminMenuCollapsibleItemClassKeys = "root" | "childSelected" | "listItem" | "open";
+
+const styles = () =>
+    createStyles<CometAdminMenuCollapsibleItemClassKeys, any>({
+        root: {},
+        childSelected: {},
+        listItem: {},
+        open: {},
+    });
+
+export interface MenuLevel {
+    level?: 1 | 2;
```

Comment:
that only supports 2 levels; should we enforce that restriction somehow?

Reply:
Probably a good idea, do you know, how we could do that?  
Currenty an error is only thrown during runtime when rendering the menu.

Reply:
Unfortunately, preventing nesting of `MenuCollapsibleItem` in the `children` prop doesn’t seem possible with typescript. I think a runtime error is safe enough anyway, also I doubt anyone would want nesting here anyway.